### ISSUE TITLE
more latex problems with references inside parsed-literal directive (``\DUrole``)

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#3207]
The latex writer uses ``\DUrole`` with a comma separated first argument, but if this is found inside  `alltt` environment, the comma is TeX active and this breaks latex compilation.

```rst
.. _mysection:

Section
-------

.. parsed-literal::

   public :ref:`mysection`  antidiag()
```

latex file contains

```latex
\begin{alltt}
public {\hyperref[index:mysection]{\sphinxcrossref{\DUrole{std,std-ref}{Section}}}}  antidiag()
\end{alltt}
```

and the problem comes from the comma inside `\DUrole`. Reported at https://github.com/sphinx-doc/sphinx/issues/2026#issuecomment-265873107.

**Comment by jfbu** [comment]
Not only the comma in `\DUrole{std,std-ref}` but also the `-` is problematic. This is why I opted for `\DUrole{\detokenize{std,std-ref}}` approach by latex writer at #3208 .

**Comment by jfbu** [comment]
e9c472c fixes this by moving `\detokenize` insertion to inside `\DUrole` LaTeX macro definition itself.

Analogous issue #3200 fixed at #3206 can't do the same because there would be many core LaTeX macros to patch: `\hyperref`, `\ref`, `\pageref`, `nameref`, `\bibitem`, `\cite`... whic is definitely to avoid. Hence #3206 inserts `\detokenize` explicitely in latex output.

**Comment by jfbu** [comment]
fixed at  db97f37 (#3208)
